docs/standard/io/pipelines.md

```diff
@@ -0,0 +1,339 @@
+---
+title: "I/O pipelines - .NET"
+description: Learn how to efficiently use I/O pipelines in .NET and avoid problems in your code.
+ms.date: "10/01/2019"
+ms.technology: dotnet-standard
+helpviewer_keywords:
+  - "Pipelines"
+  - "Pipelines I/O"
+  - "I/O [.NET], Pipelines"
+author: rick-anderson
+ms.author: riande
+---
+# System.IO.Pipelines in .NET
+
+<xref:System.IO.Pipelines> is a new library that is designed to make it easier to do high-performance I/O in .NET. It’s a library targeting .NET Standard that works on all .NET implementations.
+
+<a name="solve"></a>
+
+## What problem does System.IO.Pipelines solve
+<!-- corner case doesn't MT (machine translate)   -->
+Apps that parse streaming data are composed of boilerplate code having many specialized and unusual code flows. The boilerplate and special case code is complex and difficult to maintain.
+
+`System.IO.Pipelines` was architected to:
+
+* Have high performance parsing streaming data.
+* Reduce code complexity.
+
+The following code is typical for a TCP server that receives line-delimited messages (delimited by `'\n'`) from a client:
+
+```csharp
+async Task ProcessLinesAsync(NetworkStream stream)
+{
+    var buffer = new byte[1024];
+    await stream.ReadAsync(buffer, 0, buffer.Length);
+    
+    // Process a single line from the buffer
+    ProcessLine(buffer);
+}
+```
+
+The preceding code has several problems:
+
+* The entire message (end of line) might not be received in a single call to `ReadAsync`.
+* It's ignoring the result of `stream.ReadAsync`. `stream.ReadAsync` returns how much data was read.
+* It doesn't handle the case where multiple lines are read in a single `ReadAsync` call.
```

Comment:
I'd also consider mentioned that the method is always allocating a byte array each time we read data.

Reply:
> I'd also consider mentioning that the method is always allocating a byte array each time we read data.

Added

* It allocates a `byte` array with each read.